src/averageAges.js

```diff
@@ -14,12 +20,18 @@
  *
  * @return {number}
  */
+const MALE_SEX = 'm';
+const FEMALE_SEX = 'f';
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const mensOfCenturyArray = people
+    .filter(person => person.sex === MALE_SEX)
+    .filter(person => century
+      ? century === Math.ceil(person.died / 100) : true);
+
+  return mensOfCenturyArray
+    .reduce((year, person) => year + person.died - person.born, 0)
+    / mensOfCenturyArray.length;
```

Comment:
It's not clear what exactly the function returns.
you can split it into a few variables to make it more clear.

Reply:
and in other functions too